scripts/stablex/ensure_owl_liquidity.js

```diff
@@ -0,0 +1,92 @@
+const BatchExchange = artifacts.require("BatchExchange")
+const { decodeOrdersBN } = require("../../src/encoding.js")
+const BN = require("bn.js")
+const { sendTxAndGetReturnValue } = require("../../test/utilities.js")
+const { fetchTokenInfo } = require("./utilities")
+const maxUint32 = new BN(2).pow(new BN(32)).sub(new BN(1))
+
+const MINIMAL_LIQUIDITY_FOR_OWL = new BN(10).pow(new BN(17))
+const SELL_ORDER_AMOUNT_OWL = new BN(10).pow(new BN(18)).mul(new BN(5))
+const OWL_NUMBER_DIGITS = 18
+// All orders provided by this liquidity script  will sell OWL for a very high price:
+// At 1000 [token]/[OWL]. In most of the cases this will ensure that 1 [OWL] is valued
+// higher than 1 dollar. For tokens valued below 1/10000 USD, OWL can be extracted profitably
+// from these orders. But since we only sell 5 OWL and 10 OWL have to be spent to add one token,
+// stealing OWL by adding new tokens will not be profitable.
+const PRICE_FOR_LIQUIDITY_PROVISION = new BN(10000)
+
+const isOneSellOrderProvidingLiquidity = function(orders) {
+  for (const order of orders) {
+    if (order.sellTokenBalance.gt(MINIMAL_LIQUIDITY_FOR_OWL) && order.remainingAmount.gt(MINIMAL_LIQUIDITY_FOR_OWL)) {
+      return true
+    }
+  }
+  return false
+}
+
+// This function checks whether it is likely that there is a liquidity provision order from Gnosis
+// in the set of the orders. It does so by looking at two order criteria: SellAmount and validUntil.
+// While this is really just a heuristic check, it should be sufficient for now.
+const hasOWLLiquidityOrderAlreadyBeenPlaced = function(orders) {
+  for (const order of orders) {
+    if (order.priceDenominator.eq(SELL_ORDER_AMOUNT_OWL) && order.validUntil == maxUint32.toNumber()) {
+      return true
+    }
+  }
+  return false
+}
+
+const sendLiquidityOrder = async function(instance, tokenId) {
```

Comment:
This method could be unit tested to document the logic and make sure it's working. Maybe it can return just the parameters of the orders which can be collected and then issued in a place multiple order tx?